### PR TITLE
[Schema] Support TensorOrTupleTensor

### PR DESCRIPTION
**Pull request by comaniac**
## Description ##
Per offline discussion with @zhen-jia and investigation, we put the idea of merging cast and group_cast together away for now. The reason is that although we could allow `cast(data, "float16")` and `cast([data1, data2], "float16")`, the output type of these two ops are different (tensor type and tuple type), which results in incompatibility of existing passes. Given that `group_cast` is currently only used by ZeRO and fixing all passes is impractical and complicate, we decided to keep `group_cast` for now.

On the other hand, if we will have another non-collective group op in the future, we should definitely find a systematic solution.

Meanwhile, this PR still makes some improvements to the schema system: When the argument schema is TensorTuple, support the argument type to be a tensor or a list of tensors.

## Checklist ##

- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

cc @awslabs/raf-reviewer @zhen-jia @yzhliu

### CODE REVIEW

File: src/op/regs/ffi2expr.h [hunk 1]
```diff
@@ -241,6 +241,32 @@ inline ir::Expr TupleTensor(const registry::TVMArgValue& a) {
   throw;
 }
 
+inline ir::Expr TensorOrTupleTensor(const registry::TVMArgValue& a) {
```
Comment (hgt312) [inline src/op/regs/ffi2expr.h:244]: remove the `TupleTensor` above?

File: src/op/regs/ffi2schema.h [hunk 2]
```diff
@@ -258,6 +258,37 @@ inline std::vector<value::BaseTensorValue> TupleTensor(const registry::TVMArgVal
   throw;
 }
 
+inline std::vector<value::BaseTensorValue> TensorOrTupleTensor(const registry::TVMArgValue& a) {
```
Comment (hgt312) [inline src/op/regs/ffi2schema.h:261]: ditto

File: src/op/regs/value2schema.h [hunk 3]
```diff
@@ -209,6 +209,31 @@ inline std::vector<value::BaseTensorValue> TupleTensor(const value::Value& a) {
   throw;
 }
 
+inline std::vector<value::BaseTensorValue> TensorOrTupleTensor(const value::Value& a) {
```
Comment (hgt312) [inline src/op/regs/value2schema.h:212]: ditto

**Review by hgt312 — APPROVED** [review]
LGTM

File: src/op/regs/ffi2expr.h (hunk 1 above)
Comment (comaniac) [reply src/op/regs/ffi2expr.h:221]: Good catch. Removed.

**Comment by comaniac** [comment]
Thanks @hgt312

**Pull request merged**